spec/lcm/integration/lcm_end_to_end_spec.rb

```diff
@@ -0,0 +1,255 @@
+      end
+    end
+  end
+
+  describe '2 - Initial Provisioning' do
+    before(:all) do
+      @config_template_path = File.expand_path(
+        '../params/provisioning_brick.json.erb',
+        __FILE__
+      )
+    end
+
+    include_context 'provisioning brick'
+
+    it 'creates client projects only for filtered segments' do
+      expect($client_projects.length).to be 2
+    end
+
+    it_behaves_like 'a synchronization brick' do
+      let(:original_project) { $master }
+      let(:projects) { $client_projects }
+      let(:schedules_status) { 'ENABLED' }
+      let(:lcm_managed_tag) { true }
+      let(:update_operations) { [] }
+      let(:update_scripts) { [] }
+      let(:client_id_schedule_parameter) { true }
+      let(:user_group) { false }
+      let(:schedule_diff) do
+        [['+', 'hiddenParams.hidden_msg_from_provisioning_brick', nil],
+         ['+', 'params.msg_from_provisioning_brick', 'Hi, I was set by provisioning brick']]
+      end
+    end
+  end
+
+  describe '3 - Initial Rollout' do
+    before(:all) do
+      @config_template_path = File.expand_path(
+        '../params/rollout_brick.json.erb',
+        __FILE__
+      )
+    end
+
+    include_context 'rollout brick'
+
+    it_behaves_like 'a synchronization brick' do
+      let(:original_project) { $master }
+      let(:projects) { $client_projects }
+      let(:schedules_status) { 'ENABLED' }
+      let(:lcm_managed_tag) { true }
+      let(:update_operations) do
+        [
+          {
+            'updateOperation' => {
+              'type' => 'dataset.drop',
+              'dataset' => 'dataset.quotes',
+              'destructive' => true,
+              'description' => "Drop dataset '%s'",
+              'parameters' => ['Stock Quotes Data']
+            }
+          }
+        ]
+      end
+      let(:update_scripts) do
+        [
+          {
+            'updateScript' => {
+              'maqlDdl' => "DROP ALL IN {dataset.quotes} CASCADE;\n",
+              'maqlDdlChunks' => ["DROP ALL IN {dataset.quotes} CASCADE;\n"],
+              'preserveData' => true,
+              'cascadeDrops' => true
+            }
+          },
+          {
+            'updateScript' => {
+              'maqlDdl' => "DROP ALL IN {dataset.quotes} CASCADE;\n",
+              'maqlDdlChunks' => ["DROP ALL IN {dataset.quotes} CASCADE;\n"],
+              'preserveData' => false,
+              'cascadeDrops' => true
+            }
+          }
+        ]
+      end
+      let(:client_id_schedule_parameter) { true }
+      let(:user_group) { false }
+      let(:schedule_diff) do
+        [['+', 'hiddenParams.hidden_msg_from_rollout_brick', nil],
+         ['+', 'params.msg_from_rollout_brick', 'Hi, I was set by rollout brick']]
+      end
+    end
+  end
+
+  describe '4 - Modify development project' do
+    it 'modifies development project' do
+      skip 'bug'
+      @project.schedules.each(&:disable!)
+    end
+  end
+
+  describe '5 - Subsequent Release' do
```

Comment:
Similar code found in lcm_end_to_end_spec.rb:20, lcm_end_to_end_spec.rb:140 (mass = 124)